src/GafferML/TensorToImage.cpp

```diff
@@ -0,0 +1,329 @@
+{
+	ConstStringVectorDataPtr channels = channelsPlug()->getValue()->copy();
+	// `channels` might be in a non-standard order, to facilitate unpacking from
+	// a shuffled buffer, and could contain duplicates since it's easy to create
+	// them while shuffling the list in the UI. Sort into a more natural order
+	// and remove duplicates.
+	StringVectorDataPtr result = new StringVectorData( ImageAlgo::sortedChannelNames( channels->readable() ) );
+	result->writable().erase(
+		std::unique(
+			result->writable().begin(),
+			result->writable().end()
+		),
+		result->writable().end()
+	);
+	// Remove empty channel names since they wouldn't be valid in the output.
+	result->writable().erase(
+		std::remove_if(
+			result->writable().begin(),
+			result->writable().end(),
+			[] ( const string &channelName ) {
+				return channelName.empty();
+			}
+		),
+		result->writable().end()
+	);
+	return result;
+}
+
+void TensorToImage::hashChannelData( const GafferImage::ImagePlug *parent, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	FlatImageSource::hashChannelData( parent, context, h );
+	{
+		ImagePlug::GlobalScope globalScope( context );
+		tensorPlug()->hash( h );
+		channelsPlug()->hash( h );
+		interleavedChannelsPlug()->hash( h );
+	}
+
+	h.append( context->get<string>( ImagePlug::channelNameContextName ) );
+	h.append( context->get<V2i>( ImagePlug::tileOriginContextName ) );
+}
+
+IECore::ConstFloatVectorDataPtr TensorToImage::computeChannelData( const std::string &channelName, const Imath::V2i &tileOrigin, const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	ConstTensorPtr tensorData;
+	ConstStringVectorDataPtr channelsData;
+	bool interleavedChannels;
+	{
+		ImagePlug::GlobalScope globalScope( context );
+		tensorData = tensorPlug()->getValue();
+		channelsData = channelsPlug()->getValue();
+		interleavedChannels = interleavedChannelsPlug()->getValue();
+	}
+
+	const auto channelIt = std::find( channelsData->readable().begin(), channelsData->readable().end(), channelName );
+	if( channelIt == channelsData->readable().end() )
+	{
+		throw IECore::Exception( fmt::format( "Invalid channel \"{}\"", channelName ) );
+	}
+	const size_t channelIndex = channelIt - channelsData->readable().begin();
+
+	const ImageShape imageShape = ::imageShape( tensorData.get(), interleavedChannels );
+	if( channelIndex >= imageShape.numChannels )
+	{
+		throw IECore::Exception( fmt::format( "Channel \"{}\" out of range", channelName ) );
+	}
+
+	FloatVectorDataPtr outData = new FloatVectorData;
+	vector<float> &out = outData->writable();
+
+	const Box2i dataWindow = imageShape.dataWindow;
+	const Box2i tileBound( tileOrigin, tileOrigin + V2i( ImagePlug::tileSize() ) );
+	const Box2i validTileBound = BufferAlgo::intersection( dataWindow, tileBound );
+	out.resize( ImagePlug::tileSize() * ImagePlug::tileSize() );
+
+	const float *sourceData = tensorData->value().GetTensorData<float>();
```

Comment:
Oh, no, I have not! I think it would crash right now, at least for the `FLOAT16` case. Should I throw if it isn't float? Or do some templating to support the conversion? And if the latter, do I need to worry about integer formats too?

Reply:
I haven't tested a model where the current code wouldnt work but I was considering that for my `TensorToMesh`.
We can worry about that later.